### PR TITLE
Convert SymbolSearchOptions to editorconfig options

### PR DESCRIPTION
**Pull request by tmat**
Changes `dotnet_search_reference_assemblies` to a supported editorconfig option (used by VS Code) and `dotnet_unsupported_search_nuget_packages` to an unsupported one.

`SymbolSearchOptions` were the last options on `CodeActionOptions`. Removing the former allows us to remove the latter.

### CODE REVIEW

File: src/Features/Core/Portable/AddMissingReference/AbstractAddMissingReferenceCodeFixProvider.cs [hunk 1]
```diff
@@ -20,16 +20,6 @@ namespace Microsoft.CodeAnalysis.AddMissingReference;
 
 internal abstract partial class AbstractAddMissingReferenceCodeFixProvider : AbstractAddPackageCodeFixProvider
 {
-    /// <summary>
-    /// Values for these parameters can be provided (during testing) for mocking purposes.
-    /// </summary> 
-    protected AbstractAddMissingReferenceCodeFixProvider(
-        IPackageInstallerService? packageInstallerService = null,
-        ISymbolSearchService? symbolSearchService = null)
-        : base(packageInstallerService, symbolSearchService)
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/AddMissingReference/AbstractAddMissingReferenceCodeFixProvider.cs:29]: oh nice.

File: src/Features/Core/Portable/AddPackage/AbstractAddPackageCodeFixProvider.cs [hunk 2]
```diff
@@ -18,25 +16,14 @@
 
 namespace Microsoft.CodeAnalysis.AddPackage;
 
+/// <summary>
+/// Values for parameters can be provided (during testing) for mocking purposes.
+/// </summary> 
 internal abstract partial class AbstractAddPackageCodeFixProvider : CodeFixProvider
 {
-    private readonly IPackageInstallerService _packageInstallerService;
-    private readonly ISymbolSearchService _symbolSearchService;
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/AddPackage/AbstractAddPackageCodeFixProvider.cs:24]: oh.  you fixed this.  nice!

File: src/Features/Core/Portable/AddPackage/AbstractAddPackageCodeFixProvider.cs [hunk 3]
```diff
@@ -46,31 +33,35 @@ internal abstract partial class AbstractAddPackageCodeFixProvider : CodeFixProvi
 
         var workspaceServices = document.Project.Solution.Services;
 
-        var symbolSearchService = _symbolSearchService ?? workspaceServices.GetService<ISymbolSearchService>();
-        var installerService = _packageInstallerService ?? workspaceServices.GetService<IPackageInstallerService>();
+        if (workspaceServices.GetService<ISymbolSearchService>() is not { } symbolSearchService ||
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/AddPackage/AbstractAddPackageCodeFixProvider.cs:36]: fwiw, i don't like `not { } name` to mix the retrival and null check.  i'd prefer a distinct null check step.

File: src/Features/Core/Portable/AddPackage/AbstractAddSpecificPackageCodeFixProvider.cs [hunk 4]
```diff
@@ -17,10 +17,7 @@ internal abstract partial class AbstractAddSpecificPackageCodeFixProvider : Abst
     /// <summary>
     /// Values for these parameters can be provided (during testing) for mocking purposes.
     /// </summary> 
-    protected AbstractAddSpecificPackageCodeFixProvider(
-        IPackageInstallerService packageInstallerService = null,
-        ISymbolSearchService symbolSearchService = null)
-        : base(packageInstallerService, symbolSearchService)
+    protected AbstractAddSpecificPackageCodeFixProvider()
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/AddPackage/AbstractAddSpecificPackageCodeFixProvider.cs:20]: feeel free to remove.

File: src/Features/Core/Portable/SymbolSearch/SymbolSearchOptions.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+    [DataMember]
+    public bool SearchNuGetPackages { get; init; } = true;
+
+    public static readonly SymbolSearchOptions Default = new();
+}
+
+internal static class SymbolSearchOptionsStorage
+{
+    private static readonly OptionGroup s_optionGroup = new(name: "symbol_search", description: FeaturesResources.Symbol_search);
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/SymbolSearch/SymbolSearchOptions.cs:27]: maybe dotnet_symbol_search?  what's the namign we're generally using for these groups?

File: src/Features/DiagnosticsTestUtilities/CodeActions/CSharpCodeFixVerifier`2+Test.cs [hunk 6]
```diff
@@ -109,35 +102,6 @@ protected override CompilationOptions CreateCompilationOptions()
                 return compilationOptions.WithSpecificDiagnosticOptions(compilationOptions.SpecificDiagnosticOptions.SetItems(CSharpVerifierHelper.NullableWarnings));
             }
 
-#if !CODE_STYLE
```
Comment (CyrusNajmabadi) [inline src/Features/DiagnosticsTestUtilities/CodeActions/CSharpCodeFixVerifier`2+Test.cs:112]: awesome.

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Features/Core/Portable/SymbolSearch/SymbolSearchOptions.cs (hunk 5 above)
Comment (tmat) [reply src/Features/Core/Portable/SymbolSearch/SymbolSearchOptions.cs:27]: "dotnet" is part of the option name. It ends up as `dotnet.symbolSearch.searchReferenceAssemblies` in VS Code after the transformations we do.

**Pull request merged**